File: frontend/src/app/services/movimiento.service.ts
import { inject, Injectable } from '@angular/core';
import { environment } from '../../enviroments/enviroment';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Movimiento } from '../models/movimiento.model';

@Injectable({
  providedIn: 'root',
})
export class MovimientoService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/movimientos`;

  listar(): Observable<Movimiento[]> {
    return this.http.get<Movimiento[]>(this.apiUrl);
  }

  listarPorCuenta(cuentaId: number): Observable<Movimiento[]> {
    return this.http.get<Movimiento[]>(`${this.apiUrl}/cuenta/${cuentaId}`);
  }

  crear(movimiento: Movimiento): Observable<Movimiento> {
    return this.http.post<Movimiento>(this.apiUrl, movimiento);
  }
}

File: backend/src/main/java/com/portfolio/banco/service/MovimientoService.java
package com.portfolio.banco.service;

import java.util.List;

import com.portfolio.banco.dto.movimiento.MovimientoRequestDTO;
import com.portfolio.banco.dto.movimiento.MovimientoResponseDTO;

public interface MovimientoService {

    MovimientoResponseDTO crear(MovimientoRequestDTO dto);

    List<MovimientoResponseDTO> listar();

    List<MovimientoResponseDTO> listarPorCuenta(Long cuentaId);
}
